src/main/java/net/unicon/lti/controller/lti/DeepLinkController.java

```diff
@@ -0,0 +1,133 @@
+ */
+package net.unicon.lti.controller.lti;
+
+import com.google.common.hash.Hashing;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import net.unicon.lti.exceptions.ConnectionException;
+import net.unicon.lti.model.lti.dto.DeepLinkJWTDTO;
+import net.unicon.lti.model.lti.dto.DeepLinkRequest;
+import net.unicon.lti.model.lti.dto.NonceState;
+import net.unicon.lti.service.lti.DeepLinkService;
+import net.unicon.lti.service.lti.LTIJWTService;
+import net.unicon.lti.service.lti.NonceStateService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * This Deep Link controller will return the signed JWT with the deeplinks selected in
+ * the deep link dialog.
+ */
+@Controller
+@Scope("session")
+@RequestMapping("/deeplink")
+public class DeepLinkController {
+
+    static final Logger log = LoggerFactory.getLogger(DeepLinkController.class);
+
+    @Autowired
+    LTIJWTService ltijwtService;
+
+    @Autowired
+    DeepLinkService deepLinkService;
+
+    @Autowired
+    NonceStateService nonceStateService;
+
+
+    @RequestMapping({"/toJwt"})
+    public ResponseEntity<Object> deepLinksToJwt(@RequestBody DeepLinkRequest deeplinksRequested) throws ConnectionException, GeneralSecurityException, IOException {
+        NonceState nonceState = nonceStateService.getNonce(deeplinksRequested.getNonce());
+        ResponseEntity<Object> responseEntity = checkAccess(deeplinksRequested, nonceState);
+
+        if (responseEntity != null) {
+            return responseEntity;
+        }
+        Jws< Claims> stateClaims = ltijwtService.validateState(nonceState.getState());
+        Jws< Claims> idToken = ltijwtService.validateJWT(deeplinksRequested.getId_token(), stateClaims.getBody().get("clientId", String.class));
+        if (deeplinksRequested.getSelectedIds().isEmpty()){
+            Map<String, Object> errorResponse = new HashMap<>();
+            errorResponse.put("error", "Not Items selected");
+            errorResponse.put("message", "Empty list of items");
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(errorResponse);
+        }
+        log.info(String.join(",", deeplinksRequested.getSelectedIds()));
+        DeepLinkJWTDTO JSONdeeplink = deepLinkService.generateDeepLinkJWT(deeplinksRequested.getSelectedIds(), idToken);
+
+        return ResponseEntity.ok(JSONdeeplink);
+    }
+
+    @RequestMapping({"/deleteNonce"})
```

Comment:
Should this be either `@PostMapping` or `@DeleteMapping` instead of `@RequestMapping`? I think `@DeleteMapping` might be ideal but then I don't think you could have a body anymore.

Reply:
If I change it it will need to be a POST... yes, because I need the body to check the security.